src/DotNetty.Common/Utilities/StringUtil.cs

```diff
@@ -0,0 +1,331 @@
+    /// String utility class.
+    /// </summary>
+    public static class StringUtil
+    {
+        public static readonly string Newline;
+
+        public const char DoubleQuote = '\"';
+        public const char Comma = ',';
+        public const char LineFeed = '\n';
+        public const char CarriageReturn = '\r';
+        public const char Tab = '\t';
+
+        public const byte UpperCaseToLowerCaseAsciiOffset = 'a' - 'A';
+        static readonly string[] Byte2HexPad = new string[256];
+        static readonly string[] Byte2HexNopad = new string[256];
+
+        /**
+         * 2 - Quote character at beginning and end.
+         * 5 - Extra allowance for anticipated escape characters that may be added.
+        */
+        static readonly int CsvNumberEscapeCharacters = 2 + 5;
+        static readonly char PackageSeparatorChar = '.';
+
+        static StringUtil()
+        {
+            // Determine the newline character of the current platform.
+            string newLine;
+            try
+            {
+                //newLine = new Formatter().format("%n").toString();
+                newLine = Environment.NewLine;
+            }
+            catch
+            {
+                // Should not reach here, but just in case.
+                newLine = "\n";
+            }
+
+            Newline = newLine;
+
+            // Generate the lookup table that converts a byte into a 2-digit hexadecimal integer.
+            int i;
+            for (i = 0; i < 10; i++)
+            {
+                var buf = new StringBuilder(2);
+                buf.Append('0');
+                buf.Append(i);
+                Byte2HexPad[i] = buf.ToString();
+                Byte2HexNopad[i] = (i).ToString();
+            }
+            for (; i < 16; i++)
+            {
+                var buf = new StringBuilder(2);
+                char c = (char)('a' + i - 10);
+                buf.Append('0');
+                buf.Append(c);
+                Byte2HexPad[i] = buf.ToString();
+                Byte2HexNopad[i] = c.ToString(); /* String.valueOf(c);*/
+            }
+            for (; i < Byte2HexPad.Length; i++)
+            {
+                var buf = new StringBuilder(2);
+                buf.Append(i.ToString("X") /*Integer.toHexString(i)*/);
+                string str = buf.ToString();
+                Byte2HexPad[i] = str;
+                Byte2HexNopad[i] = str;
+            }
+        }
+
+        /// <summary>
+        /// Splits the specified {@link String} with the specified delimiter in maxParts maximum parts.
+        /// This operation is a simplified and optimized
+        /// version of {@link String#split(String, int)}.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="delim"></param>
+        /// <param name="maxParts"></param>
+        /// <returns></returns>
+        public static string[] Split(string value, char delim, int maxParts)
+        {
+            int end = value.Length;
```

Comment:
in porting stage that we are right now, it makes sense to keep things as they are in netty. That said, at one point we may further integrate with .NET std routines etc. so please, collect suggestions like this and file a batch issue for them.